src/lib/react-query/projects.ts

```diff
@@ -517,3 +517,54 @@ export function setAutostopDataSyncTimeoutMutationOptions({
 		},
 	} satisfies UseMutationOptions<void, Error, { after: number | null }>
 }
+
+export function exportGeoJSONMutationOptions({
+	projectApi,
+}: {
+	projectApi: MapeoProjectApi
+}) {
+	return {
+		...baseMutationOptions(),
+		mutationFn: async (opts) => {
+			return projectApi.exportGeoJSONFile(opts.path, opts.exportOptions)
+		},
+	} satisfies UseMutationOptions<
+		string,
+		Error,
+		{
+			path: string
+			exportOptions: {
+				observations: boolean
+				tracks: boolean
+				lang: string
+			}
+		}
+	>
+}
+
+export function exportZipFileMutationOptions({
+	projectApi,
+}: {
+	projectApi: MapeoProjectApi
+}) {
+	return {
+		...baseMutationOptions(),
+		mutationFn: async (opts) => {
+			return projectApi.exportZipFile(opts.path, {
+				...opts.exportOptions,
+				attachments: true,
```

Comment:
opted to just make attachments true here because the only use case for the API is when attachments are involved. If there are no attachments the user should be using exportGeoJSON, not this (even though technically they still could). If this changes in the future, were only really using it one place, so i don't think it will be hard to change

Reply:
Not too keen on this. would rather this be an explicit thing that's provided by where the hook is used.

Reply:
in general, I'd prefer that core-react is a flexible wrapper around core. consumers of this module should define their own wrapping hooks to make it more opinionated